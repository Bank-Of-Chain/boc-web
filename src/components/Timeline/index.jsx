import React from "react"
import classNames from "classnames"

import { makeStyles } from "@material-ui/core/styles"
import Timeline from "@material-ui/lab/Timeline"
import TimelineItem from "@material-ui/lab/TimelineItem"
import TimelineSeparator from "@material-ui/lab/TimelineSeparator"
import TimelineContent from "@material-ui/lab/TimelineContent"
import TimelineDot from "@material-ui/lab/TimelineDot"
import TimelineOppositeContent from "@material-ui/lab/TimelineOppositeContent"
import TimelineCard from "../TimelineCard"

import styles from "./style"

const useStyles = makeStyles(styles)

export default function CustomTimeline(props) {
  const {
    events = [],
    topText = 'Now',
    firstPlace = 'right',
    cardColorReverse = false,
    connectorColorRevese = false
  } = props
  const classes = useStyles()
  const firstInLeft = firstPlace !== 'right'
  const displayEvent = [...events]
  if (firstInLeft) {
    displayEvent.unshift({
      date: '',
      event: 'placeholder'
    })
  }

  return (
    <div className={classes.timelineWrapper}>
      <div className={classes.topInfo}>{topText}</div>
      <div className={classes.events}>
        <div
          className={classNames(classes.connector, {
            [classes.connectorColorRevese]: connectorColorRevese
          })}
        />
        <Timeline className={classes.timeline} align="alternate">
          {displayEvent.map((item, index) => {
            const isOdd = index % 2 === 0
            const cardProps = {
              colorReverse: cardColorReverse
            }
            if (!isOdd) {
              cardProps.arrowDirection = "right"
            }
            const hidden = firstInLeft && index === 0
            return (
              <TimelineItem key={item.event} className={classNames(classes.timelineItem, { [classes.hidden]: hidden })}>
                <TimelineOppositeContent>
                  <p className={classes.keyTime}>{item.date}</p>
                </TimelineOppositeContent>
                <TimelineSeparator className={classes.separator}>
                  <TimelineDot className={classes.dot} />
                </TimelineSeparator>
                <TimelineContent>
                  <TimelineCard {...cardProps}>{item.event}</TimelineCard>
                </TimelineContent>
              </TimelineItem>
            )
          })}
        </Timeline>
      </div>
    </div>
  )
}
