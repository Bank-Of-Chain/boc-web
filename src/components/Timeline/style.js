const style = {
  timelineWrapper: {
    position: 'relative',
    width: '100%'
  },
  topInfo: {
    width: '278px',
    height: '56px',
    lineHeight: '56px',
    border: '1px solid #A68EFE',
    margin: '0 auto',
    textAlign: 'center',
    borderRadius: '4px',
    fontSize: '24px',
    fontWeight: 700,
    color: '#A68EFE'
  },
  events: {
    position: 'relative',
    width: '100%'
  },
  connector: {
    position: 'absolute',
    left: '50%',
    width: '1px',
    height: '100%',
    transform: 'translate(-50%, 0)',
    background: 'linear-gradient(0, rgba(32, 32, 35, 0) 0%, #BEBEBE 100%)'
  },
  connectorColorRevese: {
    background: 'linear-gradient(180deg, rgba(32, 32, 35, 0) 0%, #BEBEBE 100%)'
  },
  timeline: {
    padding: '46px 16px',
    margin: 0
  },
  timelineItem: {
    margin: '28px 0'
  },
  hidden: {
    display: 'none'
  },
  keyTime: {
    position: 'relative',
    top: '50%',
    margin: 0,
    color: '#A0A0A0',
    fontSize: '20px',
    lineHeight: '30px',
    transform: 'translate(0, -50%)'
  },
  separator: {
    margin: '0 24px'
  },
  dot: {
    position: 'relative',
    top: '50%',
    width: '6px',
    height: '6px',
    margin: 0,
    background: '#A0A0A0',
    transform: 'translate(0, -50%)'
  },
  '@media (max-width: 768px)': {
    separator: {
      marginLeft: '8px',
      marginRight: '8px'
    }
  }
}

export default style
