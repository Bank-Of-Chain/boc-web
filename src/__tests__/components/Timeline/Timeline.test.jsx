import React from 'react'
import { render } from '@testing-library/react'
import Timeline from '@/components/Timeline'

test('Timeline Component Render', () => {
  const { asFragment } = render(<Timeline />)
  expect(asFragment()).toMatchSnapshot()
})
