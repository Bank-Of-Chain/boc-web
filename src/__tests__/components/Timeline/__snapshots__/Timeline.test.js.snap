// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline Component Render 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-timelineWrapper-1"
  >
    <div
      class="makeStyles-topInfo-2"
    >
      Now
    </div>
    <div
      class="makeStyles-events-3"
    >
      <div
        class="makeStyles-connector-4"
      />
      <ul
        class="MuiTimeline-root MuiTimeline-alignAlternate makeStyles-timeline-6"
      />
    </div>
  </div>
</DocumentFragment>
`;
